awx/main/analytics/collectors.py

```diff
@@ -282,32 +283,43 @@ def _copy_table(table, query, path):
 
 @register('events_table', '1.2', format='csv', description=_('Automation task records'), expensive=True)
 def events_table(since, full_path, until, **kwargs):
-    events_query = '''COPY (SELECT main_jobevent.id, 
+    # The events_table collector will no longer respect the value of
+    # `since`, as that would potentially cause gaps in the output
+    # under circumstances where continuous output is desired.
+
+    truncate = now() - timedelta(weeks=4)
+    truncate_pk = models.JobEvent.objects.filter(created__lt=truncate).aggregate(Max('pk'))['pk__max']
+    previous_pk = Setting.objects.filter(key='AUTOMATION_ANALYTICS_LAST_EVENT').first()  # Bypass cache
+    previous_pk = max(previous_pk or 0, truncate_pk or 0)
+    final_pk = models.JobEvent.objects.filter(id__gt=previous_pk, created_lte=until).aggregate(Max('pk'))['pk__max']
+    settings.AUTOMATION_ANALYTICS_LAST_EVENT = final_pk
```

Comment:
This implies the copy is successful, and will cause a skip of data if it's not.

Reply:
True enough.  I'll see if I can work around that.

Reply:
I've brought it forward after the `_copy_table` call.

Reply:
That may not be sufficient if we fail to send the data.